packages/react-debug-tools/src/__tests__/ReactHooksInspectionIntegration-test.js

```diff
@@ -335,6 +337,46 @@ describe('ReactHooksInspectionIntegration', () => {
     ]);
   });
 
+  it('should inject legacy context which is not a hook', () => {
+    class LegacyContextProvider extends React.Component<any> {
+      static childContextTypes = {
+        string: PropTypes.string,
+      };
+
+      getChildContext() {
+        return {
+          string: 'abc',
+        };
+      }
+
+      render() {
+        return this.props.children;
+      }
+    }
+
+    function FunctionalLegacyContextConsumer(props, context) {
+      return <div>{context.string}</div>;
+    }
+    FunctionalLegacyContextConsumer.contextTypes = {
+      string: PropTypes.string,
+    };
+    const renderer = ReactTestRenderer.create(
+      <LegacyContextProvider>
+        <FunctionalLegacyContextConsumer />
+      </LegacyContextProvider>,
+    );
+
+    const childFiber = renderer.root
+      .findByType(FunctionalLegacyContextConsumer)
+      ._currentFiber();
+    const hasLegacyContext = !!childFiber.elementType.contextTypes;
+    if (hasLegacyContext) {
```

Comment:
Why is the test doing this check? Seems kind of like lower-level implementation details, and could also lead to a false positive (e.g. replace with `!!childFiber.elementType.contextTypess` above and the test would still pass because the boolean check would just fail silently).

Since we *know* the legacy context is being used here, why do we need a conditional at all?

Reply:
That was my probably inappropriate attempt to prevent test from running on versions where context should be deprecated.
Removed a conditional & added value verification.